pkg/state/types.go

```diff
@@ -672,26 +673,59 @@ func (e EmptyInterfaceUsingUnderlyingType) Less(i, j int) bool {
 	return strings.Compare(objI, objJ) == -1
 }
 
-// Helper function to sort nested arrays in a map
-func sortNestedArrays(m map[string]interface{}) map[string]interface{} {
-	sortedMap := make(map[string]interface{})
+// Helper function to get schema for a field name
+func getSchemaForFieldName(schema gjson.Result, fieldName string) gjson.Result {
+	fieldsQuery := "fields.#(" + fieldName + ")." + fieldName
+	result := schema.Get(fieldsQuery)
+	if !result.Exists() {
+		// try shorthand fields
+		shorthandQuery := "shorthand_fields.#(" + fieldName + ")." + fieldName
+		result = schema.Get(shorthandQuery)
+	}
+	return result
+}
+
+// Helper function to determine if we should skip sorting based on schema
+func skipSort(schema gjson.Result) bool {
```

Comment:
Suggestion:
Instead of skipSort that negates the outcome of this function during usage, what if we use `shouldSort` instead? Makes the code flow more natural. WDYT?

Reply:
Makes sense 👍